### PR TITLE
[LLM] Upgrade lm_eval integration and usage

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change
enhancement, evaluation part upgrade and change usages.
@VincyZhang  Please check the CI issue.
## Description

lm_eval recommends wrapper model based on the `LM` class usage, so `HFCausalLM`, `HFSeq2SeqLM` is created. 
update the lm_eval commit to latest, `2c18e367c6ded428863cd1fd4cf9558ca49d68dc`
update usage
pytorch
```
from intel_extension_for_transformers.llm.evaluation.lm_eval import HFCausalLM, HFSeq2SeqLM, evaluate
clm_tokenizer =  AutoTokenizer.from_pretrained("facebook/opt-125m")
clm_model = HFCausalLM(model=clm_model, tokenizer=clm_tokenizer)
seq2seq_tokenizer = AutoTokenizer.from_pretrained("t5-small")
seq2seq_model = HFCausalLM(model=seq2seq_model, tokenizer=seq2seq_tokenizer)
results = evaluate(
    model=clm_model,
    tasks=["piqa"],
    batch_size=1,
    limit=20,
    no_cache=True
)
results = evaluate(
    model=seq2seq_model,
    tasks=["piqa"],
    batch_size=1
    limit=20,
    no_cache=True
)
```
onnx
```
from intel_extension_for_transformers.llm.evaluation.lm_eval import HFCausalLM, HFSeq2SeqLM, evaluate
model = HFCausalLM(model_name_or_path="./gptj", model_format="onnx")   
results = evaluate(
    model= model,
    tasks=["piqa"],
    limit=20,
    no_cache=True
)

model = HFSeq2SeqLM(model_name_or_path="./t5-past", model_format="onnx")
results = evaluate(
    model=model,
    tasks=["piqa"],
    limit=20,
    no_cache=True
)
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/llm/evaluation/lm_eval/lm_model.py [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+                'Args model is necessary when model_format="torch". \n'
+                + "the following is the correct usage, \n"
+                + "model = HFCausalLM(model=model, tokenizer=tokenizer) or "
+                + "model = HFSeq2SeqLM(model=model, tokenizer=tokenizer)"
+            )
+            exit()
+
+        if self.model_format == "onnx" and model_name_or_path is None:
+            logger.error(
+                'Args model is necessary when model_format="torch". \n'
```
[the remark this answers is not in the document]
Comment (changwangss) [reply intel_extension_for_transformers/llm/evaluation/lm_eval/lm_model.py]: Thanks review, improved.

**Comment by changwangss** [comment]
"format scan" failed due to finetuning.py but I didn't modify the file, please check again. @VincyZhang

**Comment by changwangss** [comment]
IPEX install not correct, because torch is 2.1 but ipex is 2.0, please check @VincyZhang

**Comment by VincyZhang** [comment]
> "format scan" failed due to finetuning.py but I didn't modify the file, please check again. @VincyZhang

please check your code here: https://github.com/intel/intel-extension-for-transformers/pull/409/files#diff-3ab0e2db773fcf388060aaf05220000ebeeada0c5b0a21980d5971bfb2a37835L118
and also, for langchain not import, you can refer to this PR https://github.com/intel/intel-extension-for-transformers/pull/416/files

**Comment by VincyZhang** [comment]
> IPEX installation is not correct, because torch is 2.1 but ipex is 2.0, please check @VincyZhang

IPEX released, we can have a try 
https://github.com/intel/intel-extension-for-pytorch/releases/tag/v2.1.0.dev%2Bcpu.llm.mlperf

**Review by xin3he — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]
LGTM. Are you going to create a separate PR for lm harness readme?

**Comment by VincyZhang** [comment]
@changwangss NeuralChat UT finetune part failed, please check

**Comment by changwangss** [comment]
> @changwangss NeuralChat UT finetune part failed, please check
improved. it failed due to neuralchat lm_eval version.

**Comment by changwangss** [comment]
> LGTM. Are you going to create a separate PR for lm harness readme?

added it.

**Comment by wenhuach21** [comment]
is it possible to simply the usage and users only need to give the model and we wrapper it automatically?

**Comment by changwangss** [comment]
> is it possible to simply the usage and users only need to give the model and we wrapper it automatically?

causal seq2seq may be easy to distinguish, but there are tokenizer, backend and other information that also need to be filled in by the user. I think the complexity is the same.

**Comment by changwangss** [comment]
open #617 , close this pr.

**Pull request closed without merging**